tests/filecheck/backend/convert_riscv_scf_to_riscv_cf.mlir

```diff
@@ -6,7 +6,7 @@ builtin.module {
         %zero = riscv.li 0 : !riscv.reg<a2>
         %step = riscv.li 4 : !riscv.reg<a3>
         %forty = riscv.li 40 : !riscv.reg<a4>
-        riscv_scf.for %offset : !riscv.reg<a5> = %zero to %forty step %step {
+        riscv_scf.for %offset = %zero to %forty step %step : !riscv.reg<a5> {
```

Comment:
This brings the format inline with `scf.for` where the induction variable type is printed last when it does not default to `index` type.

Reply:
wait but this is different, the offset is not the step

Reply:
The same thing happens with `scf.for`

Reply:
I'm worried this will be confusing. This is less confusing in MLIR, as all the types are the same, but in regalloc'ed riscv_scf, each of lb, ub, and step may be in different registers. Also actually we may want the lowering to riscv_cf to add a move from lb to induction, meaning that each of these would have a separate type. We may want to discuss that first before changing the API here?

Reply:
I understand the confusing bit, but I fail to see what it has to do lowering or optimizing it.
This is just the print format, nothing has changed in the underlying representation of the for operation.
What am I missing?

As a stop gap, the location of the induction variable printing can also be an option.

Reply:
what's changed is that the type is currently only for the induction variable, and is printed near it. the proposed change implies that it's either the type of all the variables, or the step one specifically